### PR TITLE
Fall back to using Segment integration when we cannot get settings

### PR DESCRIPTION
**Pull request by tonyxiao**
cc @f2prateek @ladanazita 

To repro the original buggy behavior
1) Put device in airplane mode and open app
2) Send an event `Application Launched 1`
3) Force kill the app via task switcher
4) Disable airplane mode and regain internet access
5) Launch the app again, send another event `Application Launched 2`
6) Observe that `Application Launched 2` will hit tracking API, while `Application Launched 1` will be missing. 

Basically what happens is that during step 2, if the settings request were to fail, no integrations would be initialized (segment included). This hotfix falls back to segment integration in those cases and we'll have a more comprehensive set of fixes to address this from a design level in development branch.

### CODE REVIEW

File: Analytics.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = "Analytics"
-  s.version          = "3.5.3"
```
Comment (f2prateek) [inline Analytics.podspec]: Should stay as is.

File: Analytics/Classes/SEGAnalytics.m [hunk 2]
```diff
@@ -562,8 +562,16 @@ - (void)refreshSettings
     self.settingsRequest = [self.httpClient settingsForWriteKey:self.configuration.writeKey completionHandler:^(BOOL success, NSDictionary *settings) {
         if (success) {
             [self setCachedSettings:settings];
+        } else {
+            // Hotfix: If settings request fail, fall back to using just Segment integration
```
Comment (f2prateek) [inline Analytics/Classes/SEGAnalytics.m]: Hotfix seems like a weird comment to leave in. I'd rather put in something like "todo: we need to fix this behaviour"

File: Analytics/Classes/SEGAnalytics.m [hunk 3]
```diff
@@ -583,7 +591,7 @@ + (void)debug:(BOOL)showDebugLogs
 
 + (NSString *)version
 {
-    return @"3.5.3";
```
Comment (f2prateek) [inline Analytics/Classes/SEGAnalytics.m]: Should stay as (same as above) and updated as part of the release process.

**Comment by tonyxiao** [comment]
@f2prateek updated

**Comment by f2prateek** [comment]
@tonyxiao nice, Let's do a quick manual test to verify the offline behaviour and get this out (adding one more comment but shouldn't block the change)

File: Analytics/Classes/SEGAnalytics.m [hunk 4]
```diff
@@ -562,8 +562,16 @@ - (void)refreshSettings
     self.settingsRequest = [self.httpClient settingsForWriteKey:self.configuration.writeKey completionHandler:^(BOOL success, NSDictionary *settings) {
         if (success) {
             [self setCachedSettings:settings];
+        } else {
+            // TODO: If settings request fail, fall back to using just Segment integration
+            // Won't catch situation where this callback never gets called - that will get addressed separately in regular dev 
+            [self setCachedSettings:@{
+                @"integrations": @{
+                    @"Segment.io": @{ @"apiKey": self.configuration.writeKey },
```
Comment (f2prateek) [inline Analytics/Classes/SEGAnalytics.m]: Eventually instead of doing setCachedSettings we should do updateIntegrationWithSettings (can't do this yet because the code references self.cachedSettings directly in a few places)>

**Comment by tonyxiao** [comment]
@f2prateek Yea I did a manual test and verified that it behaves as expected. Merging.

File: Analytics/Classes/SEGAnalytics.m (hunk 4 above)
Comment (tonyxiao) [inline Analytics/Classes/SEGAnalytics.m]: Agreed. Wanted to change as few lines as possible since this is a hotfix. It'll become a lot more clear in dev.
